backend/agentd/middlewares.go

```diff
@@ -0,0 +1,129 @@
+package agentd
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	"github.com/sensu/sensu-go/backend/apid/middlewares"
+	"github.com/sensu/sensu-go/backend/authentication/jwt"
+	"github.com/sensu/sensu-go/backend/authorization"
+	"github.com/sensu/sensu-go/backend/authorization/rbac"
+	"github.com/sensu/sensu-go/backend/store"
+	"github.com/sensu/sensu-go/transport"
+)
+
+// AuthenticationMiddleware represents the middleware used for authentication
```

Comment:
Simon, Nikki and I discussed the rationale for these global variables on Slack. They are to support patching from the enterprise product.

I do think we should include a TODO here about what we intend to do to support graceful degradation in the future.

Reply:
@echlebek I'll document the enterprise codebase with the TODO and some links to issues.